src/Form/Control/ScopeBuilder.php

```diff
@@ -0,0 +1,611 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\ui\Form\Control;
+
+use atk4\data\Field;
+use atk4\data\Model;
+use atk4\data\Model\Scope;
+use atk4\ui\Exception;
+use atk4\ui\Form\Control;
+use atk4\ui\Template;
+
+class ScopeBuilder extends Control
+{
+    /** @var bool Do not render label for this input. */
+    public $renderLabel = false;
+
+    /**
+     * General or field type specific options.
+     *
+     * @var array
+     */
+    public $options = [
+        'enum' => [
+            'limit' => 250,
+        ],
+        'debug' => false, // displays query output live on the page if set to true
+    ];
+    /**
+     * Max depth of nested conditions allowed.
+     * Corresponds to VueQueryBulder maxDepth.
+     * Maximum support by js component is 10.
+     *
+     * @var int
+     */
+    public $maxDepth = 3;
```

Comment:
at least 5 please and then see if more is normally needed - user should never be limited!